File: Array with Odd Sum.py
for _ in range(int(input())):
    n = int(input())
    arr = list(map(int, input().split()))
    if sum(arr) % 2 != 0:
        print("YES")
    else:
        odd = 0
        even = 0
        for i in range(n):
            if odd >= 1 and even >= 1:
                break
            if arr[i] % 2 == 0:
                even += 1
            else:
                odd += 1
        if odd >= 1 and even >= 1:
            print("YES")
        else:
            print("NO")

File: Array with Odd Sum.cpp
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--) {
        int n, even = 0, odd = 0;
        cin >> n;
        vector <int> arr;
        int temp;
        for (int i = 0; i < n; i++) {
            cin >> temp;
            if ((temp % 2) == 0) {
                even++;
            }
            else {
                odd++;
            }
            arr.push_back(temp);
        }
        int sume = 0;
        if ((even >= 1 && odd >= 1) || (accumulate(arr.begin(), arr.end(), sume) % 2 == 1)){
            cout << "YES" << endl;
        }
        else {
            cout << "NO" << endl;
        }
    }
    return 0;
}
